lib/IRGen/GenDecl.cpp

```diff
@@ -2419,7 +2420,16 @@ static llvm::GlobalVariable *createGOTEquivalent(IRGenModule &IGM,
                                       llvm::GlobalValue::PrivateLinkage,
                                       global,
                                       llvm::Twine("got.") + globalName);
-  gotEquivalent->setUnnamedAddr(llvm::GlobalValue::UnnamedAddr::Global);
+  
+  // rdar://problem/50968433: Unnamed_addr constants appear to get emitted
+  // with incorrect alignment by the LLVM JIT in some cases. Don't use
+  // unnamed_addr as a workaround.
+  if (!IGM.getOptions().UseJIT) {
+    gotEquivalent->setUnnamedAddr(llvm::GlobalValue::UnnamedAddr::Global);
+  } else {
+    gotEquivalent->setLinkage(llvm::GlobalValue::LinkOnceODRLinkage);
```

Comment:
Please use `ApplyIRLinkage` here.  Without the COMDAT, this will break the Windows build.

Reply:
It probably wouldn't matter in this case, because there'd never be any time we feed two `-use-jit`-built modules into LINK simultaneously. In immediate mode there's only one module, and in integrated-REPL mode, the REPL does its own ad-hoc coalescing and linking of separate modules.